src/runtime_src/hip/core/memory.cpp

```diff
@@ -103,7 +103,7 @@ namespace xrt::core::hip
         assert(src_hip_mem->get_flags() == hipHostMallocDefault || src_hip_mem->get_flags() == hipHostMallocPortable);
     }
 
-    const unsigned char *src_ptr = reinterpret_cast<const unsigned char *>(src);
+    const auto *src_ptr = reinterpret_cast<const unsigned char *>(src);
```

Comment:
```suggestion
    auto src_ptr = reinterpret_cast<const unsigned char *>(src);
```
Already const and ptr per cast.

Reply:
Updated